dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/telnet/InvokeTelnetHandler.java

```diff
@@ -99,13 +97,10 @@ private static boolean isMatch(Method method,List<Object> args, Class<?>[] param
                 if (!ReflectUtils.isCompatible(type, arg)) {
                     return false;
                 }
-            } else if (arg instanceof Map) {
-                String name = (String) ((Map<?, ?>) arg).get("class");
-                Class<?> cls = arg.getClass();
-                if (name != null && name.length() > 0) {
-                    cls = ReflectUtils.forName(name);
-                }
-                if (!type.isAssignableFrom(cls)) {
+            } else if (arg instanceof JSONObject) {
```

Comment:
```java
@Data
public class User1 implements Serializable {
    private static final long serialVersionUID = -4729079781904156102L;
    private String name;
}
```
```java
@Data
public class User2 implements Serializable {
    private static final long serialVersionUID = -2112637973209566052L;
    private String name;
}
```
```java
public interface XXXService{
String hello(User1 user);
String hello(User2 user);

}
```
May find not the correct overload method; I think  must **class** param when the method is overloaded.